### PR TITLE
Added events to show and hide the preview pane.

### PR DESCRIPTION
**Pull request by nbianca**
The plugin-outlet defined in the preview is useless if the preview pane is hidden.

I consider that plugins should may be able to show and hide the preview pane.

### CODE REVIEW

**Comment by SamSaffron** [comment]
Do we need an `off` for these events when you hide the composer? I don't want to keep the event listening around if the composer is not visible on the page (or even if the composer is minimized)

**Comment by nbianca** [comment]
I made the edits.

**Comment by SamSaffron** [comment]
One last style thing, I think it is better to have a central spot for changing key value store as opposed to having it in 3 places. 

I would just observe when showPreview changes and amend it based of that. It means there will only be one spot that amends the store which makes the code more robust.

**Comment by SamSaffron** [comment]
excellent, thanks!

File: app/assets/javascripts/discourse/components/composer-editor.js.es6 [hunk 1]
```diff
@@ -42,6 +50,11 @@ export default Ember.Component.extend({
     return showPreview ? I18n.t('composer.hide_preview') : I18n.t('composer.show_preview');
   },
 
+  @observes('showPreview')
```
Comment (techAPJ) [inline app/assets/javascripts/discourse/components/composer-editor.js.es6]: This resulted in a failing test. Fixed via: https://github.com/discourse/discourse/commit/3fb1c21dd5fbc17fda9e4c75953d19ed8009798e

:)

**Pull request merged**